reducers/plugins/index.js

```diff
@@ -135,9 +173,12 @@ function components(state = {}, action) {
     switch (action.type) {
     case ActionTypes.RECEIVED_PLUGIN_COMPONENT: {
         if (action.name && action.data) {
+            if (action.name === 'PostDropdownMenu' && action.data.parentId) {
+                action.data = buildSubMenu(state, action);
+            }
             const nextState = {...state};
             const currentArray = nextState[action.name] || [];
-            const nextArray = [...currentArray];
+            const nextArray = [...currentArray.filter((c) => !c.subMenu)];
```

Comment:
i think it'll be cleaner if we have the `buildSubMenu` just return a value of `null` and then this function returns `state` i.e. the original `state` value without any changes to indicate error/noop state. Thoughts?

Reply:
Excellent idea.  It really seems easier to reason about.  done.